src/supabase/helpers/tables/pretty-logs.ts

```diff
@@ -0,0 +1,185 @@
+  }
+
+  private _formatStackTrace(stack: string, linesToRemove = 0, prefix = ""): string {
+    const lines = stack.split("\n");
+    for (let i = 0; i < linesToRemove; i++) {
+      lines.shift(); // Remove the top line
+    }
+    return lines
+      .map((line) => `${prefix}${line.replace(/\s*at\s*/, "  ↳  ")}`) // Replace 'at' and prefix every line
+      .join("\n");
+  }
+
+  private _isEmpty(obj: Record<string, any>) {
+    return !Reflect.ownKeys(obj).some((key) => typeof obj[String(key)] !== "function");
+  }
+
+  private _log(type: PrettyLogsWithOk, message: any) {
+    const defaultSymbols: Record<PrettyLogsWithOk, string> = {
+      fatal: "×",
+      ok: "✓",
+      error: "⚠",
+      info: "›",
+      debug: "››",
+      verbose: "💬",
+    };
+
+    const symbol = defaultSymbols[type];
+
+    // Formatting the message
+    const messageFormatted = typeof message === "string" ? message : util.inspect(message, { showHidden: true, depth: null, breakLength: Infinity });
+    // const messageFormatted =
+    //   typeof message === "string" ? message : JSON.stringify(Logs.convertErrorsIntoObjects(message));
+
+    // Constructing the full log string with the prefix symbol
+    const lines = messageFormatted.split("\n");
+    const logString = lines
+      .map((line, index) => {
+        // Add the symbol only to the first line and keep the indentation for the rest
+        const prefix = index === 0 ? `\t${symbol}` : `\t${" ".repeat(symbol.length)}`;
+        return `${prefix} ${line}`;
+      })
+      .join("\n");
+
+    const fullLogString = logString;
+
+    const colorMap: Record<PrettyLogsWithOk, [keyof typeof console, Colors]> = {
+      fatal: ["error", Colors.fgRed],
+      ok: ["log", Colors.fgGreen],
+      error: ["warn", Colors.fgYellow],
+      info: ["info", Colors.dim],
+      debug: ["debug", Colors.fgMagenta],
+      verbose: ["debug", Colors.dim],
+    };
+
+    const _console = console[colorMap[type][0] as keyof typeof console] as (...args: string[]) => void;
+    if (typeof _console === "function") {
+      _console(this._colorizeText(fullLogString, colorMap[type][1]));
+    } else {
+      throw new Error(fullLogString);
+    }
+  }
+}
+interface Metadata {
+  error?: { stack?: string };
+  stack?: string;
+  message?: string;
+  name?: string;
+  [key: string]: any;
+}
+
+enum Colors {
+  reset = "\x1b[0m",
+  bright = "\x1b[1m",
+  dim = "\x1b[2m",
+  underscore = "\x1b[4m",
+  blink = "\x1b[5m",
+  reverse = "\x1b[7m",
+  hidden = "\x1b[8m",
+
+  fgBlack = "\x1b[30m",
+  fgRed = "\x1b[31m",
+  fgGreen = "\x1b[32m",
+  fgYellow = "\x1b[33m",
+  fgBlue = "\x1b[34m",
+  fgMagenta = "\x1b[35m",
+  fgCyan = "\x1b[36m",
+  fgWhite = "\x1b[37m",
+
+  bgBlack = "\x1b[40m",
+  bgRed = "\x1b[41m",
+  bgGreen = "\x1b[42m",
+  bgYellow = "\x1b[43m",
+  bgBlue = "\x1b[44m",
+  bgMagenta = "\x1b[45m",
+  bgCyan = "\x1b[46m",
+  bgWhite = "\x1b[47m",
+}
+export enum LogLevel {
```

Comment:
I'm impressed. It looks like you correctly took the latest code which I updated as recently as yesterday!

Reply:
Yes, I took the latest code, glad to hear that!